internal/provider/resource_deployment.go

```diff
@@ -427,15 +427,15 @@ func resourceDeploymentCreate(ctx context.Context, sub subi.SubstrateExt, d *sch
 	return &deployer, deployer.Deploy(ctx, sub)
 }
 
-func resourceDeploymentRead(ctx context.Context, sub subi.SubstrateExt, d *schema.ResourceData, apiClient *apiClient) (Marshalable, error) {
+func resourceDeploymentRead(ctx context.Context, sub subi.Substrate, d *schema.ResourceData, apiClient *apiClient) (Marshalable, error) {
 	deployer, err := getDeploymentDeployer(d, apiClient)
 	if err != nil {
 		return nil, err
 	}
 	return &deployer, nil
 }
 
-func resourceDeploymentUpdate(ctx context.Context, sub subi.SubstrateExt, d *schema.ResourceData, apiClient *apiClient) (Marshalable, error) {
+func resourceDeploymentUpdate(ctx context.Context, sub subi.Substrate, d *schema.ResourceData, apiClient *apiClient) (Marshalable, error) {
 	if d.HasChange("node") {
```

Comment:
this if block isn't needed any more `ForceNew` in the `capacity_id` field in schema is enough